src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/role/RoleAssignmentService.java

```diff
@@ -51,4 +74,77 @@ private RoleAssignmentResource getRoles(String idamUserId, String authToken) {
+        RoleAssignmentResource roleAssignmentResponse = performSearch(buildQueryForAutoAssignment(taskResource));
+        log.debug(
+            "Roles successfully retrieved from RoleAssignment Service for taskId '{}' and caseId '{}'",
+            taskResource.getTaskId(),
+            taskResource.getCaseId()
+        );
+
+        return roleAssignmentResponse.getRoleAssignmentResponse();
+    }
+
+    public List<RoleAssignment> getRolesByUserId(String userId) {
+        requireNonNull(userId, "userId cannot be null");
+
+        RoleAssignmentResource roleAssignmentResponse = roleAssignmentServiceApi.getRolesForUser(
+            userId,
+            systemUserIdamToken.generate(),
+            serviceAuthTokenGenerator.generate()
+        );
+
+        return roleAssignmentResponse.getRoleAssignmentResponse();
+    }
+
+    public RoleAssignmentResource performSearch(MultipleQueryRequest multipleQueryRequest) {
+        try {
+            return roleAssignmentServiceApi.queryRoleAssignments(
+                systemUserIdamToken.generate(),
+                serviceAuthTokenGenerator.generate(),
+                multipleQueryRequest
+            );
+        } catch (FeignException ex) {
+            throw new ServerErrorException(
+                "Could not retrieve role assignments when performing the search", ex);
+        }
+    }
+
+    private MultipleQueryRequest buildQueryForAutoAssignment(TaskResource taskResource) {
+        Set<TaskRoleResource> rolesInTask = taskResource.getTaskRoleResources();
+
+        String caseId = taskResource.getCaseId();
+        List<Classification> securityClassifications =
+            evaluateEqualOrHigherClassification(taskResource.getSecurityClassification());
+
+        Set<String> roleNamesFound = rolesInTask.stream()
+            .map(role -> role.getOwn() && role.getAutoAssignable() ? role.getRoleName() : null)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toSet());
+
+        QueryRequest queryRequest = QueryRequest.builder()
+            .roleName(new ArrayList<>(roleNamesFound))
+            .classification(securityClassifications)
+            .grantType(singletonList(GrantType.SPECIFIC))
+            .validAt(LocalDateTime.now())
+            .attributes(Map.of("caseId", List.of(caseId)))
+            .build();
+
+        return MultipleQueryRequest.builder().queryRequests(singletonList(queryRequest)).build();
+    }
+
+    private List<Classification> evaluateEqualOrHigherClassification(SecurityClassification securityClassification) {
+        switch (securityClassification) {
+            case PUBLIC:
+                return asList(PUBLIC, PRIVATE, RESTRICTED);
+            case PRIVATE:
+                return asList(PRIVATE, RESTRICTED);
+            case RESTRICTED:
+                return singletonList(RESTRICTED);
+            default:
+                throw new IllegalStateException("Unexpected classification value " + securityClassification);
```

Comment:
coverage?

Reply:
as discussed since its an enum could not cover this, and moreover this is existing piece of code, have already improved coverage